posts/2019-09-27-Async-await-hits-beta.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: post
+title: "Async-await hits beta!"
+author: Niko Matsakis
+---
+
+Big news! As of this writing, **syntactic support for async-await is
+available in the Rust beta channel!** It will be available in the 1.39
+release, which is expected to be released on **November 7th, 2019**.
+Once async-await hits stable, that will mark the culmination of a
+**multi-year effort to enable efficient and ergonomic asynchronous I/O
+in Rust**. It will not, however, mark the end of the road: there is
+still more work to do, both in terms of polish (some of the error
+messages we get today are, um, [not great]) and in terms of feature
+set ([async fn in traits], anyone?).
+
+[not great]: https://github.com/rust-lang/rust/issues/64130
+[async fn in traits]: https://github.com/dtolnay/async-trait
+
+(If you're not familiar with what async-await is, don't despair!
+There's a primer and other details later on in this post!)
+
+### Async-await support in the ecosystem growing rapidly
+
+But async-await has never been the entire story. To make good use of
+async-await, you also need strong libraries and a vibrant ecosystem.
+**Fortunately, you've got a lot of good choices, and they keep getting
+better:** 
+
+- the async runtime [tokio], for example, recently announced an [alpha
+  release][] that supports async-await;
+- the [recently announced][] [async-std][] library was built from the
+  start on the new async-await syntax;
+- using [wasm-bindgen-futures], you can even bridge Rust Futures with
+  [JavaScript promises];
+- finally, in addition to the core runtimes just mentioned,
+  async-await support is starting to become available in higher-level
+  [web frameworks][wf] as well.
+  
+[wasm-bindgen-futures]: https://docs.rs/crate/wasm-bindgen-futures/0.2.16
+[tokio]: https://actix.rs/
+[actix]: https://actix.rs/
+[alpha release]: https://tokio.rs/blog/2019-08-alphas/
+[adding support]: https://github.com/actix/actix-web/issues/955#issuecomment-523266936
+[async-std]: https://async.rs/
+[recently announced]: https://async.rs/blog/announcing-async-std/
+[wf]: https://www.arewewebyet.org/topics/frameworks/
+[JavaScript promises]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises
+
+### Restructuring Async I/O in the Rust org
+
+Now that async-await is stable, we are taking the opportunity to make
+some changes to our Rust team structure. The current structure
+includes two working groups: the [Async Foundations WG], focused on
+building up core language support, and the [Async Ecosystem WG],
+focused on supporting the ecosystem develop. 
+
+**In light of all the activity going on in the ecosystem, however,
```

Comment:
Reminder: I had planned to open an issue on the Governance WG discussing WG teardown procedures and cite that issue here.

Reply:
Done